### PR TITLE
LPS-188387 Hide M:M relationship option in SystemObjectDefinition Address

### PR DESCRIPTION
**Pull request by murilostodolni**
Related issue: https://liferay.atlassian.net/browse/LPS-188387

### CODE REVIEW

**Review by alinedoleron — APPROVED** [review]
LGTM

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsRelationshipsDisplayContext.java [hunk 1]
```diff
@@ -210,6 +212,19 @@ public String getRESTContextPath(ObjectDefinition objectDefinition) {
 		return jaxRsApplicationDescriptor.getRESTContextPath();
 	}
 
+	public boolean isAddress() {
+		ObjectDefinition objectDefinition = getObjectDefinition();
+
+		if (StringUtil.equals(
+				objectDefinition.getName(), Address.class.getSimpleName()) &&
+			objectDefinition.isUnmodifiableSystemObject()) {
+
```
[the remark this answers is not in the document]
Comment (murilostodolni) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsRelationshipsDisplayContext.java:221]: Ok, i will invert the condition.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java [hunk 2]
```diff
@@ -151,96 +152,90 @@ public void testAddObjectRelationship() throws Exception {
...
-					ObjectRelationshipConstants.TYPE_MANY_TO_MANY +
-						" does not allow a parameter object field ID",
-				objectRelationshipParameterObjectFieldIdException.getMessage());
-		}
+				ObjectRelationshipConstants.TYPE_MANY_TO_MANY));
 
-		try {
-			_objectRelationshipLocalService.addObjectRelationship(
+		AssertUtils.assertFailure(
+			ObjectRelationshipParameterObjectFieldIdException.class,
```
[the remark this answers is not in the document]
Comment (murilostodolni) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:185]: Done

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java [hunk 3]
```diff
@@ -1079,6 +1080,17 @@ private void _validate(
...
+				objectDefinition1.getName(), Address.class.getSimpleName()) ||
+			 Objects.equals(
+				 objectDefinition2.getName(), Address.class.getSimpleName())) &&
+			Objects.equals(
+				type, ObjectRelationshipConstants.TYPE_MANY_TO_MANY)) {
+
+			throw new ObjectRelationshipTypeException(
+				"Many to many relationship are not allowed with Address");
+		}
+
```
[the remark this answers is not in the document]
Comment (murilostodolni) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:1093]: Done

**Comment by murilostodolni** [comment]
ci:test:sf

**Comment by murilostodolni** [comment]
ci:test:relevant

**Comment by murilostodolni** [comment]
ci:test:object

**Review by seltong — APPROVED** [review]
LGTM

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsRelationshipsDisplayContext.java [hunk 4]
```diff
@@ -210,6 +212,19 @@ public String getRESTContextPath(ObjectDefinition objectDefinition) {
 		return jaxRsApplicationDescriptor.getRESTContextPath();
 	}
 
+	public boolean isAddress() {
```
Comment (carolmariaabb) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsRelationshipsDisplayContext.java:215]: @murilostodolni can we rename it to `isAddressSystemObject`? `isAddress` seems so generic..

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsRelationshipsDisplayContext.java [hunk 5]
```diff
@@ -210,6 +212,19 @@ public String getRESTContextPath(ObjectDefinition objectDefinition) {
...
+		ObjectDefinition objectDefinition = getObjectDefinition();
+
+		if (objectDefinition.isUnmodifiableSystemObject() &&
+			StringUtil.equals(
+				objectDefinition.getName(), Address.class.getSimpleName())) {
```
Comment (carolmariaabb) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsRelationshipsDisplayContext.java:220]: @murilostodolni can we use the `objectDefinition.getClassName()` instead?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java [hunk 6]
```diff
@@ -1079,6 +1080,17 @@ private void _validate(
 			}
 		}
 
+		if (Objects.equals(
+				type, ObjectRelationshipConstants.TYPE_MANY_TO_MANY) &&
+			(Objects.equals(
+				objectDefinition1.getName(), Address.class.getSimpleName()) ||
+			 Objects.equals(
+				 objectDefinition2.getName(), Address.class.getSimpleName()))) {
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:1088]: @murilostodolni can we use the `objectDefinition.getClassName()` instead?

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsRelationshipsDisplayContext.java (hunk 5 above)
Comment (murilostodolni) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsRelationshipsDisplayContext.java:220]: Yes, we can. I'm changing this.

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsRelationshipsDisplayContext.java (hunk 4 above)
Comment (murilostodolni) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsRelationshipsDisplayContext.java:215]: Yes, I will rename to `isAddressSystemObjectDefinition`

**Comment by murilostodolni** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]
LGTM!

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by murilostodolni** [comment]
ci:forward:force

**Pull request closed without merging**